api/classes/base_controller.js

```diff
@@ -87,27 +87,32 @@ BaseController.prototype.getAllAsMeta = function(req, reply) {
   reply(req.generateResponse(req.pre.records.toJSON()));
 };
 
+// NOTE: creating the tarball for a project can be a lengthy process, so
+// we do it in multiple turns of the event loop, so that other requests
+// don't get blocked.
 BaseController.prototype.getAllAsTar = function(req, reply) {
   var files = req.pre.records.models;
   var tarStream = Tar.pack();
-  var self = this;
-
-  function startStreaming(files, model) {
-    return Promise.map(files, function(file) {
-      return model.query({
-        where: {
-          id: file.get('id')
-        },
-        columns: ['buffer']
-      }).fetch().then(function(model) {
+  var model = this.Model;
+
+  function processFile(file) {
+    return model.query({
+      where: {
+        id: file.get('id')
+      },
+      columns: ['buffer']
+    }).fetch().then(function(model) {
+      setImmediate(function() {
         tarStream.entry({ name: file.get('path') }, model.get('buffer'));
       });
-    }).then(function() {
-      tarStream.finalize();
     });
   }
 
-  startStreaming(files, self.Model).catch(errors.generateErrorResponse);
+  setImmediate(function() {
+    Promise.map(files, processFile)
```

Comment:
I think this sounds like a good idea.  Do you have a suggestion as to what I should set it to?  The docs don't really give much help here, just indicate that you can limit it.  What's a good number, do you think?